imageai/Detection/Detector.py

```diff
@@ -0,0 +1,1060 @@
+import cv2
+
+from imageai.Detection.keras_retinanet.models.resnet import resnet50_retinanet
+from imageai.Detection.keras_retinanet.utils.image import read_image_bgr, read_image_array, read_image_stream, \
+    preprocess_image, resize_image
+from imageai.Detection.keras_retinanet.utils.visualization import draw_box, draw_caption
+from imageai.Detection.keras_retinanet.utils.colors import label_color
+
+import matplotlib.pyplot as plt
+import matplotlib.image as pltimage
+import numpy as np
+import tensorflow as tf
+import os
+from keras import backend as K
+from keras.layers import Input
+from PIL import Image
+import colorsys
+
+from imageai.Detection.YOLOv3.models import yolo_main, tiny_yolo_main
+from imageai.Detection.YOLOv3.utils import letterbox_image, yolo_eval
+
+import traceback
+
+def get_session():
+    config = tf.ConfigProto()
+    config.gpu_options.allow_growth = True
+    return tf.Session(config=config)
+
+
+class ObjectDetection:
+    """
+                    This is the object detection class for images in the ImageAI library. It provides support for RetinaNet
+                     , YOLOv3 and TinyYOLOv3 object detection networks . After instantiating this class, you can set it's properties and
+                     make object detections using it's pre-defined functions.
+
+                     The following functions are required to be called before object detection can be made
+                     * setModelPath()
+                     * At least of of the following and it must correspond to the model set in the setModelPath()
+                      [setModelTypeAsRetinaNet(), setModelTypeAsYOLOv3(), setModelTypeAsTinyYOLOv3()]
+                     * loadModel() [This must be called once only before performing object detection]
+
+                     Once the above functions have been called, you can call the detectObjectsFromImage() function of
+                     the object detection instance object at anytime to obtain observable objects in any image.
+    """
+
+    def __init__(self):
+        self.__modelType = ""
+        self.modelPath = ""
+        self.__modelPathAdded = False
+        self.__modelLoaded = False
+        self.__model_collection = []
+
+        # Instance variables for RetinaNet Model
+        self.__input_image_min = 1333
+        self.__input_image_max = 800
+
+        self.numbers_to_names = {0: 'person', 1: 'bicycle', 2: 'car', 3: 'motorcycle', 4: 'airplane', 5: 'bus',
```

Comment:
rather define this as constant at the beginning of module or as constant in the class